hack/ensure-acr-login.sh

```diff
@@ -22,16 +22,12 @@ set +o xtrace
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 cd "${REPO_ROOT}" || exit 1
 
-if [[ "${REGISTRY:-}" =~ capzci\.azurecr\.io ]]; then
+# TODO(nawazkh): Why are we allowing other registries?
+if [[ "${REGISTRY:-}" =~ \.azurecr\.io ]]; then
```

Comment:
I borrowed this change from here https://github.com/kubernetes-sigs/cluster-api-provider-azure/compare/main...jsturtevant:cluster-api-provider-azure:remove-az-cli-cred-checks#diff-f43b095ca6bb739d40185ed223560171c1371512ab597e9b20d9c1fc7e308b7dR30

Do we know why we are allowing `<other-registry>.azurecr.io`, @jsturtevant ?

Reply:
for scenarios where you are running this locally and you're using your own registry

Reply:
Makes sense, just wanted to get it documented. Thank you